### ISSUE TITLE
Kommer aldrig videre end "Henter data"

### ISSUE THREAD

**Issue by mibyge** [torben87/DitElforbrugExcelPriser#3]
Hej.

Jeg har netop opdaget dit spændende regneark, og ønsker at prøve det :)

Efter at have fulgt vejledningen, og fået genereret en token samt opsat arket med de nødvendige oplysninger og tilpasninger, så forsøger jeg at opdatere data, men Excel kommer aldrig i mål. 

Jeg svarer som beskrevet på de beskeder Excel kommer med, og så hentes der en mængde data ned, men derefter sker det ikke mere, og den kommer aldrig ud af "Henter data (tryk på Esc for at annullere): <et tal>". Heller ikke selvom jeg lader den stå i en halv times tid. Herefter bliver jeg nødt til at slå excel.exe ned via Joblisten for at komme videre.

Jeg benytter Windows 10 v21H2, og Office Pro Plus 2019 v2202.

Jeg fint både se og downloade mine målerdata på eloverblik.dk, så data er til stede.

Hvad kan der være galt?

På forhånd tak.

Mvh.
Michael

**Comment by mibyge** [comment]
Jeg har prøvet at lege lidt med datoerne for perioden der hentes data for.

Enten så er problemet, at jeg ikke kan hente data større end en bestemt periode eller også er der nogle data på et bestemt tidspunkt der får det hele til at knække. Dog ser der ikke umiddelbart ud til at være nogen data der falder udenfor når jeg kigger i de data jeg har hentet via eloverblik.dk.

Lige nu ser det dog ud til, at jeg har hentet så meget data, at jeg har ramt en grænse i API'et, og dermed ikke må hente mere data lige nu, så jeg må prøve igen senere for at se om jeg kan indsnævre det yderligere.

**Comment by mibyge** [comment]
Nu er der gået en uge, og det ser stadig ikke ud til, at jeg får lov at hente data.

![image](https://user-images.githubusercontent.com/50886706/159346134-3aa68bf8-bdd0-4f00-b9a4-2e961674c562.png)

Er det noget du har oplevet?

**Comment by torben87** [comment]
Hej Michael,
Jeg har hørt fra nogle brugere, der anvender en 32 bit version af Excel, at de har problemer med at læse data ind. Jeg må være ærlig at sige, at jeg ikke har optimeret særligt meget på ram-forbruget i mine queries, da det ikke har været et problem for mig. Men jeg vil tro det er det, du oplever. Jeg ser på, om jeg kan lave nogle performanceforbedringer, og så sender jeg en ny version ud, når det er klar. Tak for dit input :)

**Comment by Gitdyr** [comment]
Jeg kører med en 64-bits version af Office. Jeg har angivet mit token og får bl.a. beskeden:

![image](https://user-images.githubusercontent.com/1284557/189519332-c7ed5df6-6221-4ec1-9d3f-8073788b31d1.png)

Hos https://www.energidataservice.dk/ skriver de, at API'et er blevet ændret. Er det det, der er problemet?

**Comment by mgades** [comment]
Ja, det tror jeg du har ret i. Det lader til at "datastore_search_sql" siden 1. september ikke længere er tilgængelig og man skal opdatere sine API kald.

https://www.energidataservice.dk/news

"You need to change the method you use for retrieving data. The datastore_search_sql method will no longer be available and the datastore_search method is changed."
